diff --git a/burger_war/scripts/seigoRun2.py b/burger_war/scripts/seigoRun2.py
index cd2afc0..d2ac995 100755
--- a/burger_war/scripts/seigoRun2.py
+++ b/burger_war/scripts/seigoRun2.py
@@ -11,6 +11,7 @@ import cv2
 import numpy as np
 import requests
 
+import rospkg
 import rospy
 import tf
 import actionlib
@@ -55,8 +56,11 @@ class SeigoBot2:
     def __init__(self):
 
         def load_waypoint():
-            path = os.environ['HOME'] + \
-                '/catkin_ws/src/burger_war/burger_war/scripts/waypoints.csv'
+            rospack=rospkg.RosPack()
+            path=os.path.join(rospack.get_path("burger_war"),"scripts","waypoints.csv")
+            print("path", path)
+            #path = os.environ['HOME'] + \
+            #    '/catkin_ws/src/burger_war/burger_war/scripts/waypoints.csv'
             return Waypoints(path, self.my_side)
 
         self.listener = tf.TransformListener()
