diff --git a/autotest/autotest.sh b/autotest/autotest.sh
index 1191401..aa1bdc2 100755
--- a/autotest/autotest.sh
+++ b/autotest/autotest.sh
@@ -89,5 +89,6 @@ do
     do_game ${i} 5 225 # 180 * 5/4
     do_game ${i} 6 225 # 180 * 5/4
     do_game ${i} 7 225 # 180 * 5/4
+    do_game ${i} 8 225 # 180 * 5/4
     #do_push
 done
diff --git a/burger_war/launch/sim_robot_run.launch b/burger_war/launch/sim_robot_run.launch
index 4507fff..d80651d 100644
--- a/burger_war/launch/sim_robot_run.launch
+++ b/burger_war/launch/sim_robot_run.launch
@@ -62,6 +62,14 @@
 	<arg name="side" value="b" />
       </include>  
     </group>
+
+    <!-- level 8 (if you want to run , set launch arg enemy_level=8 --> 
+    <group if="$(eval enemy_level==8)"> 
+      <include file="$(find burger_war_level8)/launch/your_burger.launch">
+        <arg name="robot_namespace" value="enemy_bot" />
+	<arg name="side" value="b" />
+      </include>  
+    </group>
   </group>
 
 </launch>
