From 24fbafaeba98c729fbc53789952da0caf7ad8eec Mon Sep 17 00:00:00 2001
From: seigot <you@example.com>
Date: Sat, 5 Sep 2020 16:34:23 +0000
Subject: [PATCH 4/6] update mix,max distance parameter

---
 burger_war/scripts/seigoRun2.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/burger_war/scripts/seigoRun2.py b/burger_war/scripts/seigoRun2.py
index 3847c9e..3279793 100755
--- a/burger_war/scripts/seigoRun2.py
+++ b/burger_war/scripts/seigoRun2.py
@@ -125,9 +125,9 @@ class SeigoBot2:
         self.enable_escape_approach = rospy.get_param(
             '~enable_escape_approach', default="False")
         self.escape_approach_distance_th_min = rospy.get_param(
-            '~escape_approach_distance_th_min', default=0.5)
+            '~escape_approach_distance_th_min', default=0.23)
         self.escape_approach_distance_th_max = rospy.get_param(
-            '~escape_approach_distance_th_max', default=0.5)
+            '~escape_approach_distance_th_max', default=0.85)
         self.escape_approach_time_interval = rospy.get_param(
             '~escape_approach_time_interval', default=6)
         print("enable_escape_approach", self.enable_escape_approach)
-- 
2.7.4

